### PR TITLE
MAISTRA-1937 Clean up API

### PR DESCRIPTION
**Pull request by dgn**

### CODE REVIEW

File: deploy/examples/maistra_v2_servicemeshcontrolplane_cr_minimal.yaml [hunk 1]
```diff
@@ -6,6 +6,9 @@ spec:
   tracing:
     # change to Jaeger to enable tracing
     type: None
+  techPreview:
+    wasmExtensions:
+      enabled: true
```
Comment (rcernich) [inline deploy/examples/maistra_v2_servicemeshcontrolplane_cr_minimal.yaml]: Did you mean to commit these changes?

File: manifests-maistra/2.0.0/maistraoperator.v2.0.0.clusterserviceversion.yaml [hunk 2]
```diff
@@ -12,7 +12,7 @@ metadata:
       The Maistra Operator enables you to install, configure, and manage an instance of Maistra service mesh. Maistra is based on the open source Istio project.
 
     containerImage: maistra/istio-ubi8-operator:2.0.0
-    createdAt: 2020-10-12T13:32:16EDT
+    createdAt: 2020-10-20T15:49:09CEST
```
Comment (rcernich) [inline manifests-maistra/2.0.0/maistraoperator.v2.0.0.clusterserviceversion.yaml]: Can this be dropped from the commit?

File: manifests-servicemesh/2.0.0/servicemeshoperator.v2.0.0.clusterserviceversion.yaml [hunk 3]
```diff
@@ -12,7 +12,7 @@ metadata:
       The OpenShift Service Mesh Operator enables you to install, configure, and manage an instance of Red Hat OpenShift Service Mesh. OpenShift Service Mesh is based on the open source Istio project.
 
     containerImage: registry.redhat.io/openshift-service-mesh/istio-rhel8-operator:2.0.0
-    createdAt: 2020-10-12T13:32:20EDT
+    createdAt: 2020-10-20T15:49:14CEST
```
Comment (rcernich) [inline manifests-servicemesh/2.0.0/servicemeshoperator.v2.0.0.clusterserviceversion.yaml]: Can this be dropped from the commit?

File: pkg/apis/maistra/v2/security.go [hunk 4]
```diff
@@ -235,6 +223,19 @@ type ControlPlaneSecurityConfig struct {
...
+// control plane.
+type DataPlaneSecurityConfig struct {
+	// Enable mutual TLS by default.
+	// .Values.global.mtls.enabled
+	MTLS *bool `json:"mtls,omitempty"`
+	// Auto configures the mesh to automatically detect whether or not mutual
+	// TLS is required for a specific connection.
+	// .Values.global.mtls.auto
+	// +optional
+	AutoMTLS *bool `json:"automtls,omitempty"`
```
Comment (rcernich) [inline pkg/apis/maistra/v2/security.go:236]: nit: should the json tag be `autoMTLS`?  Not sure what the standard style is for things like this.

**Review by rcernich — APPROVED** [review]
Looks good.  Just a few minor comments.  Thanks for cleaning this up!

File: deploy/examples/maistra_v2_servicemeshcontrolplane_cr_minimal.yaml (hunk 1 above)
Comment (dgn) [reply deploy/examples/maistra_v2_servicemeshcontrolplane_cr_minimal.yaml]: Nope, this was accidental. Thanks!

File: manifests-maistra/2.0.0/maistraoperator.v2.0.0.clusterserviceversion.yaml (hunk 2 above)
Comment (dgn) [reply manifests-maistra/2.0.0/maistraoperator.v2.0.0.clusterserviceversion.yaml]: I could drop it, but it's been legitimately re-generated and has new content - is the new date a problem?

File: pkg/apis/maistra/v2/security.go (hunk 4 above)
Comment (dgn) [reply pkg/apis/maistra/v2/security.go:236]: I tried that, but it looked horrible. Happy to change if we prefer it, but I thought `automtls` and `mtls` are a better fit than `autoMTLS` and `mtls`

File: manifests-maistra/2.0.0/maistraoperator.v2.0.0.clusterserviceversion.yaml (hunk 2 above)
Comment (dgn) [reply manifests-maistra/2.0.0/maistraoperator.v2.0.0.clusterserviceversion.yaml]: Oh, nevermind- the CRDs are separate. I'll remove the date changes

File: pkg/apis/maistra/v2/security.go (hunk 4 above)
Comment (rcernich) [reply pkg/apis/maistra/v2/security.go:236]: works for me.  are there any other json tags that you think we should change?  i'd just like to be somewhat consistent (and i struggle with this myself).

Comment (dgn) [reply pkg/apis/maistra/v2/security.go:236]: as discussed on IRC: not really :)

**Review by rcernich — APPROVED** [review]

File: pkg/apis/maistra/v2/smcp_new.yaml [hunk 5]
```diff
@@ -286,7 +283,8 @@ spec:
         ecdhCurves: []
         minProtocolVersion: TLSv1_2
         maxProtocolVersion: TLSv1_3
-
+    dataPlane:
+      auto: true
```
Comment (jwendell) [inline pkg/apis/maistra/v2/smcp_new.yaml]: automtls?

Comment (dgn) [reply pkg/apis/maistra/v2/smcp_new.yaml]: Good point! Forgot to fix the example

**Review by jwendell — APPROVED** [review]

**Pull request merged**